phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -137,7 +141,12 @@
 public class IndexUtil {
     public static final String INDEX_COLUMN_NAME_SEP = ":";
     public static final byte[] INDEX_COLUMN_NAME_SEP_BYTES = Bytes.toBytes(INDEX_COLUMN_NAME_SEP);
-    
+
+    private static Cache<String, Boolean> IndexNameGlobalIndexCheckerEnabledMap = CacheBuilder.newBuilder()
```

Comment:
@swaroopak Could you empty this cache when you do upgrade?

Reply:
@gokceni How? The upgrade tool would run on another machine/process. And this cache is on client-side, right?

Reply:
hmm, even if it is not possible, the cache entries expire every 30 mins.